### PR TITLE
Upgrade node, npm and specify/lock node-gyp versions.

### PR DESCRIPTION
**Pull request by vraravam**
### Description
Going to the latest version and locking for trialing building on ARM laptop.

### Checklist:

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`$ npm run lint`)
- [x] I tested/previewed my changes locally

### CODE REVIEW

**Comment by kris7t** [comment]
Do we really need exact versions? E.g., Arch only has nodejs 14.16.0 (even in `[testing]`). The package [was already flagged out of date](https://archlinux.org/packages/community/x86_64/nodejs-lts-fermium/), but will probably require a couple of days before getting updated. Nevertheless (safe for the version incompatibility error introduced by this PR), I have no problems with building Ferdi.

**Comment by vraravam** [comment]
> Do we really need exact versions? E.g., Arch only has nodejs 14.16.0 (even in `[testing]`). The package [was already flagged out of date](https://archlinux.org/packages/community/x86_64/nodejs-lts-fermium/), but will probably require a couple of days before getting updated. Nevertheless (safe for the version incompatibility error introduced by this PR), I have no problems with building Ferdi.

In which case, do you want to hold off for a few days and then merge this PR? If so, I will wait for your explicit approval before merging this.

**Comment by kris7t** [comment]
Is there something specific in 14.17.0 that we need to arm macs? In that case, I'm ok with holding this off for a bit and then merging. But looking through distro packages e.g., at https://pkgs.org/search/?q=nodejs it seems many distributions only carry 14.16, and those are likely on slower update cadences than Arch. So if we merge this, we (temporarily) exclude a bunch of distros from building Ferdi. (I only looked at distros on this list that package nodejs 14. Those on 12 are likely to use `nvm`, anyway, but it's nice to be able to rely on the native package if one is available with the correct major version.)

If 14.16 is enough, I'd be more comfortable with something like `^14.16.0` (and caret dependencies for `node-gyp` and `npm`, too, as most distros wouldn't update at _exactly_ the same time).

**Comment by vraravam** [comment]
Yes - Im ok with changing to the `^` based notations for dev/build-time-system-deps. Let me make that commit.

**Review by kris7t — APPROVED** [review]
Builds fine for me now

File: appveyor.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ cache:
 image: Visual Studio 2019
 
 environment:
-  nodejs_version: "14.16.1"
+  nodejs_version: "^14.16"
```
Comment (kris7t) [inline appveyor.yml:19]: Apparently, AppVeyor fails on this? I guess we need exact versions in `appveyor.yml` (and maybe `.travis.yml`, but that build hasn't finished yet).

**Review by kris7t — CHANGES_REQUESTED** [review]

File: appveyor.yml (hunk 1 above)
Comment (vraravam) [reply appveyor.yml]: yes - reverted to the same -tx

**Comment by vraravam** [comment]
In the end, the PR got changed to only be used to enforce the `<major>.<minor>` versions for the system-level dependencies. Upgrades have been rolled back since linux contributors might not be in a position to upgrade.

**Pull request merged**